Algorithms/Greedy/KnapsackUpdatedWithValuesOfObjects.c

```diff
@@ -0,0 +1,106 @@
+
+the problems in fractional knapsack is to
+
+-n items
+-fraction of item i is taken to maximize profit unless the bag is full. Globally optimal choice is to take item with largest profit as p_i*(weight of object/weight remaining) and Greedy choice property for maximizing profit holds.
+Dynamic Programming can solve Subproblems as well as overlapping Subproblems and finds a best solution by solving subproblems.
+
+Fractional Knapsack can be solved using both Greedy algorithm and Dynamic Programming.
+Modify this so you take a value v[i] for each item i that acts as a multiplier to the profit as:
+for fraction x_i of object i, the profit will now be: p_i * x_i * (v_i * (x_i)^2). Also, show that if the greedy solution works for this. if not, solve using DP?
+-->when with each item i the multiplier is provided v[i] then the problem is same as fractional knapsack we can arrange the items according to value per unit weight*v[i] instead of only value per unit weight .
+As it is a added on problem of fractional Knapsack ,so it can be solved using DP as well.*/
+
+int n;/*Number of objects given,a fraction object we are to choose and keep inside (x[]) knapsack having total weight  m */
+struct Object
+{	/*this is the structure defined for each object . Each 	object will have weight w, profit p, profit per unit 	weight f , and a char id(a single character name given to the object) 	assigned to object*/
+	float w;
+	float p;
+	float f;
+	float v;
+	char id;
+};
+typedef struct Object PW;
+//profit(p) and weight(w) is all that we are concerned with .
+PW *ptr;
+//initilising a pointer to object Structure
+/* The function Fractionalknapsack() gives the array showing how much fraction of each object will result an array saying what fraction of each object and total profit .
+Time Complexity ofthis function is O(n) as there is only a for loop for one by one taking objects(sorted by profit per unit weight)*/
+void Fractinalknapsack(float m)
+{
+	int i;
+	float u,x[n],profit=0.0;
+	for(i=1;i<=n;i++)
+		x[i]=0.0;
+	u=m;
+	for(i=0;i<n;i++)
+	{
+		if(u<ptr[i].w)
+			break;
+		x[i]=1;
+		u=u-ptr[i].w;
+	}
+	if(i<=n)
+		x[i]=u/ptr[i].w;
+	printf("Fraction\tId\tprofit\tWeight\n");
+	for(i=0;i<n;i++)
+	{
+		printf("%0.2f\t\t %c\t %0.2f \t %0.2f\n",x[i],ptr[i].id,(ptr[i].p)*x[i]*(ptr[i].v)*x[i]*x[i],(ptr[i].w)*x[i]);
+		profit+=(ptr[i].p)*x[i]*(ptr[i].v)*x[i]*x[i];
+	}
+	printf("\nTotal profit by keeping objects with values is %f",profit);
+
+}
+/*n objects are inputed(weight,profit of each object).Each object is given an id starting from A .
+The objects are sorted according to profit per unit weight (any sorting algorithm can be used).Here I have used Bubble Sort for Simplicity .
+Time Complexity of total program entirely depends upon the sorting algorithm as the complexity of greedy approach of Fractionalknapsack function is O(n).
+Here,total complexity is O(n^2) as the complexity is 
+O(n^2)+O(n).*/

```

Comment:
Should change it to a sorting algorithm that works in O(nlogn). Use the STL sort function for the same.